rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala

```diff
@@ -225,16 +232,148 @@ class FindExpectedReportsJdbcRepository(
             select N.nodejoinkey, N.nodeid, N.nodeconfigids, generate_subscripts(N.nodeconfigids,1) as v
             from expectedreportsnodes N
           ) as NN
-          where ${inT2(("NN.nodeid","NN.nodeconfigids[v]"), expectedNodeConfigIds)}
-          and ${notInT2(("NN.nodeid","NN.nodeconfigids[v]"), previousNodeConfigIds)}
+          where ${in("NN.nodeconfigids[v]", nodeConfigIds.map(_.version.value))}
         ) as NNN
         on E.nodejoinkey = NNN.nodejoinkey
-      """ + rulePredicate
+        ${rulePredicate}
+        ) as X
+      group by X.pkid
+      """
+
+
+      /*
+       * This one is about one order of magnitude less efficient, event
+       * with a gin index as proposed in http://stackoverflow.com/questions/28025753/looking-in-array-with-a-big-set-of-input-values
+       */
+      val query_subscript_configids2 = s"""select
+            E.pkid, NNN.nodeid, NNN.nodeconfigids
+        from expectedreports E
+        inner join expectedreportsnodes NNN
+        on E.nodejoinkey = NNN.nodejoinkey
+        where  NNN.nodeconfigids && Array[${nodeConfigIds.map(_.version.value).mkString("'", "','", "'")}]
+       ${if(filterByRules.isEmpty) "" else " and " + in("ruleid", filterByRules.map(_.value))}"""
+
+
+      def getExpectedReports(reportIds: Set[Int]): Box[Seq[ReportMapping]] = {
```

Comment:
Inception of getExpectedReports !
could you rename one of the two getExpectedReports method please ? It makes it hard to find which one is called below